sdk/spring/azure-spring-cloud-stream-binder-eventhubs/src/main/java/com/azure/spring/eventhub/stream/binder/EventHubMessageChannelBinder.java

```diff
@@ -56,12 +56,12 @@ protected MessageHandler createProducerMessageHandler(ProducerDestination destin
         handler.setSendTimeout(producerProperties.getExtension().getSendTimeout());
         handler.setSendFailureChannel(errorChannel);
         if (producerProperties.isPartitioned()) {
-            handler.setPartitionKeyExpressionString(
-                    "'partitionKey-' + headers['" + BinderHeaders.PARTITION_HEADER + "']");
+            if (producerProperties.getPartitionKeyExpression() != null) {
+                handler.setPartitionKeyExpression(producerProperties.getPartitionKeyExpression());
+            }
```

Comment:
why this?

Reply:
This is to support the configuration of the key partition expression provided by spring cloud stream.